core/divelist.c

```diff
@@ -214,21 +231,20 @@ static double calculate_cns_dive(const struct dive *dive)
 		t = sample->time.seconds - psample->time.seconds;
 		if (sample->setpoint.mbar) {
 			po2 = sample->setpoint.mbar;
-		} else {
+		} else {	// Find FiO2 as well as PO2 at mean depth of dive segment:
 			int o2 = active_o2(dive, dc, sample->time);
-			po2 = lrint(o2 * depth_to_atm(sample->depth.mm, dive));
+			po2 = lrint(o2 * depth_to_atm((int)round((sample->depth.mm + psample->depth.mm) / 2.0), dive));
```

Comment:
why do you go to float? Integer should be fine.

Reply:
Done